lua_scripts/argvenvp_misc.lua

```diff
@@ -1,18 +1,46 @@
 --
 -- Copyright (c) 2009 Nokia Corporation.
--- Licensed under MIT license
+-- Copyright (c) 2021 Jolla Ltd.
+-- License: LGPL-2.1
 --
 
 --
 -- Here is the necessary plumbing to generate argv&envp mangling
--- for miscellaneous binaries.
+-- for miscellaneous binaries and wrappers
 --
 -- See syntax from argvenvp.lua.
 --
 
+
+local dirent = require 'posix.dirent'
+local libgen = require 'posix.libgen'
+
 dpkg_architecture = {
 	name = "dpkg-architecture",
 	path_prefixes = {"/usr/bin/"},
 	remove = {"-f"},
 }
 argvmods[dpkg_architecture.name] = dpkg_architecture
+
+
+for m_index,m_name in pairs(all_modes) do
+
+   active_mapmode = m_name
+
+   for _, file in pairs(dirent.dir(session_dir.."/wrappers."..active_mapmode)) do
```

Comment:
This should work:

```suggestion
   for file in dirent.files(session_dir.."/wrappers."..active_mapmode) do
```

More occurrences elsewhere(?)

Reply:
It doesn't work with pairs. `for iterator 'for iterator' is not callable (a table value))`